subsys/compress/Kconfig

```
@@ -0,0 +1,100 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+config NRF_COMPRESS
+	bool "nRF compression/decompression library"
+
+if NRF_COMPRESS
+
+config NRF_COMPRESS_COMPRESSION
+	bool
+	help
+	  Will enable support for compression functions in library.
+
+config NRF_COMPRESS_DECOMPRESSION
+	bool "Decompression support"
+	help
+	  Will enable support for decompression functions in library.
+
+menu "Type support"
+	depends on NRF_COMPRESS_COMPRESSION || NRF_COMPRESS_DECOMPRESSION
+
+config NRF_COMPRESS_TYPE_SELECTED
+	bool
+	help
+	  Hidden option to know if at least one implmentation has been selected
+
+menuconfig NRF_COMPRESS_LZMA
+	bool "LZMA"
+	depends on NRF_COMPRESS_DECOMPRESSION
+	select NRF_COMPRESS_TYPE_SELECTED
+	help
+	  Enables LZMA support for decompression.
+
+if NRF_COMPRESS_LZMA
+
+choice NRF_COMPRESS_LZMA_VERSION
+	prompt "Version"
+	default NRF_COMPRESS_LZMA_VERSION_LZMA2
+
+config NRF_COMPRESS_LZMA_VERSION_LZMA1
+	bool "lzma1"
+
+config NRF_COMPRESS_LZMA_VERSION_LZMA2
+	bool "lzma2"
+
+endchoice
+
+endif # NRF_COMPRESS_LZMA
+
+endmenu
+
+config NRF_COMPRESS_CHUNK_SIZE
+	int "Chunk size"
+	default 2048
+	help
+	  The chunk size is the default maximum amount of data that the decompress bytes available
+	  function will return that should be provided to the decompress function, it is not a
+	  strict requirement to follow this (unless the function returns a lower value, then data
+	  of that size must be provided)
+
+config NRF_COMPRESS_MIN_MEMORY_REQUIRED
+	hex
+	default 0x26f80 if NRF_COMPRESS_DECOMPRESS_LZMA
+	default 0
+	help
+	  Hidden symbol indicating minimum buffer size for operation if operating in malloc mode.
+
+choice NRF_COMPRESS_MEMORY_TYPE
+	prompt "Memory type for buffers"
+	default NRF_COMPRESS_MEMORY_TYPE_STATIC
+
+config NRF_COMPRESS_MEMORY_TYPE_STATIC
+	bool "Static buffers"
+	help
+	  Will use static buffers to hold the data for decompression which will reduce the
+	  available memory to everything else in the application.
+
+config NRF_COMPRESS_MEMORY_TYPE_MALLOC
+	bool "Malloc"
+	depends on COMMON_LIBC_MALLOC
+	depends on (COMMON_LIBC_MALLOC_ARENA_SIZE = -1) || (COMMON_LIBC_MALLOC_ARENA_SIZE >= NRF_COMPRESS_MIN_MEMORY_REQUIRED)
+	help
+	  Will use dynamic memory allocation to hold the data for decompression, note if there is
+	  insufficient free contiguous space then decompression support will not be usable.
```

Please pay attention to comma splices. You tend to use them a lot.
```suggestion
	  Use the dynamic memory allocation to hold the data for decompression. If there is
	  insufficient free contiguous space, the decompression support will not be usable.
```

---

Fixed with minor changes